src/modules/mc_nn_control/mc_nn_control.hpp

```diff
@@ -0,0 +1,160 @@
+#include <px4_platform_common/px4_work_queue/WorkItem.hpp>
+#include <matrix/matrix/math.hpp>
+
+#include <tflite_micro/tensorflow/lite/micro/micro_mutable_op_resolver.h>
+#include <tflite_micro/tensorflow/lite/micro/micro_interpreter.h>
+#include <tflite_micro/tensorflow/lite/schema/schema_generated.h>
+
+// Include model
+#include "control_net.hpp"
+
+#include <uORB/Publication.hpp>
+#include <uORB/Subscription.hpp>
+#include <uORB/SubscriptionCallback.hpp>
+
+// Subscriptions
+#include <uORB/topics/vehicle_local_position.h>
+#include <uORB/topics/trajectory_setpoint.h>
+#include <uORB/topics/vehicle_attitude.h>
+#include <uORB/topics/vehicle_angular_velocity.h>
+#include <uORB/topics/vehicle_status.h>
+#include <uORB/topics/register_ext_component_reply.h>
+#include <uORB/topics/arming_check_request.h>
+#include <uORB/topics/parameter_update.h>
+
+// Publications
+#include <uORB/topics/actuator_motors.h>
+#include <uORB/topics/neural_control.h>
+#include <uORB/topics/register_ext_component_request.h>
+#include <uORB/topics/unregister_ext_component.h>
+#include <uORB/topics/vehicle_control_mode.h>
+#include <uORB/topics/arming_check_reply.h>
+
+using namespace time_literals; // For the 1_s in the subscription interval
+
+class MulticopterNeuralNetworkControl : public ModuleBase<MulticopterNeuralNetworkControl>, public ModuleParams,
+	public px4::WorkItem
+{
+public:
+
+	MulticopterNeuralNetworkControl();
+	~MulticopterNeuralNetworkControl() override;
+
+	/** @see ModuleBase */
+	static int task_spawn(int argc, char *argv[]);
+
+	/** @see ModuleBase */
+	static int custom_command(int argc, char *argv[]);
+
+	/** @see ModuleBase */
+	static int print_usage(const char *reason = nullptr);
+
+	/** @see ModuleBase */
+	int print_status() override;
+
+	bool init();
+
+private:
+	void Run() override;
+
+	// Functions
+	void PopulateInputTensor();
+	void PublishOutput(float *command_actions);
+	void RescaleActions();
+	int InitializeNetwork();
+	int32_t GetTime();
+	void RegisterNeuralFlightMode();
+	void UnregisterNeuralFlightMode(int8 arming_check_id, int8 mode_id);
+	void ConfigureNeuralFlightMode(int8 mode_id);
+	void ReplyToArmingCheck(int8 request_id);
+	void CheckModeRegistration();
+
+	// Subscriptions
+	uORB::SubscriptionInterval _parameter_update_sub{ORB_ID(parameter_update), 1_s};
+	uORB::Subscription _register_ext_component_reply_sub{ORB_ID(register_ext_component_reply)};
+	uORB::Subscription _arming_check_request_sub{ORB_ID(arming_check_request)};
+	uORB::Subscription _vehicle_status_sub{ORB_ID(vehicle_status)};
+	uORB::Subscription _position_sub{ORB_ID(vehicle_local_position)};
+	uORB::Subscription _trajectory_setpoint_sub{ORB_ID(trajectory_setpoint)};
+	uORB::Subscription _attitude_sub{ORB_ID(vehicle_attitude)};
+	uORB::SubscriptionCallbackWorkItem _angular_velocity_sub{this, ORB_ID(vehicle_angular_velocity)};
+
+	// Publications
+	uORB::Publication<actuator_motors_s> _actuator_motors_pub{ORB_ID(actuator_motors)};
```

Comment:
Forgot to submit: The output of the controller seems to be directly motor commands `actuator_motors` 👀